src/SickSafetyscannersRos.cpp

```diff
@@ -414,17 +416,23 @@ SickSafetyscannersRos::createLaserScanMessage(const sick::datastructure::Data& d
   for (uint32_t i = 0; i < num_scan_points; ++i)
   {
     const sick::datastructure::ScanPoint scan_point = scan_points.at(i);
-    scan.ranges[i]                                  = static_cast<float>(scan_point.getDistance()) *
-                     data.getDerivedValuesPtr()->getMultiplicationFactor() * 1e-3; // mm -> m
-    // Set values close to/greater than max range to infinity according to REP 117
-    // https://www.ros.org/reps/rep-0117.html
-    if (scan.ranges[i] >= (0.999 * m_range_max))
+    if(m_min_intensities < static_cast<double>(scan_point.getReflectivity()))
+    {
+      scan.ranges[i]                                  = static_cast<float>(scan_point.getDistance()) *
+                      data.getDerivedValuesPtr()->getMultiplicationFactor() * 1e-3; // mm -> m
+      // Set values close to/greater than max range to infinity according to REP 117
+      // https://www.ros.org/reps/rep-0117.html
+      if (scan.ranges[i] >= (0.999 * m_range_max))
+      {
+        scan.ranges[i] = std::numeric_limits<double>::infinity();
+      }
+    }
+    else
```

Comment:
For performance this might be the better solution. For readability however, it might be better to first do the ranges calculation and afterwards apply all filters.

So that the final if-clause checks for both max range and intensity.

Reply:
I agree with you that there is a more beautiful solution. However, this is a high frequency driver, so I prefer the higher performance solution

Reply:
Thnaks for your input. I suppose we can merge it like that.